src/remote-config/remote-config.ts

```diff
@@ -333,6 +322,25 @@ class ServerTemplateImpl implements ServerTemplate {
       });
   }
 
+  /**
+   * Takes in either a {@link ServerTemplateData} or a JSON string
+   * representing the template, parses it, and caches it.
+   */
+  public set(template: ServerTemplateData | string): void {
+    if (isString(template)) {
+      try {
+        this.cache = new ServerTemplateDataImpl(JSON.parse(template));
```

Comment:
Since `ServerTemplateDataImpl` throws `FirebaseRemoteConfigError` itself, I think this should be outside the try block, so we're not catching our own error, eg:

```
let parsed;
try {
  parsed = JSON.parse(template);
} catch (e) {
  // Transforms JSON parse errors to Firebase error.
  throw new FirebaseRemoteConfigError(
  'invalid-argument',
  `Failed to parse the JSON string: ${template}. ` + e);
}
// Throws template parse errors.
this.cache = new ServerTemplateDataImpl(parsed);
```

Reply:
Good idea, I needed to add an else condition to set `parsed = template`, but updated 👍